Xamarin.Forms.Core/ToolbarItem.cs

```diff
@@ -12,11 +12,34 @@ public class ToolbarItem : MenuItem
 
 		static readonly BindableProperty PriorityProperty = BindableProperty.Create("Priority", typeof(int), typeof(ToolbarItem), 0);
 
+		public static readonly BindableProperty IsVisibleProperty = BindableProperty.Create("IsVisible", typeof(bool), typeof(ToolbarItem), true, BindingMode.TwoWay, propertyChanged: (bindable, oldvalue, newvalue) =>
+		{
+			var item = bindable as ToolbarItem;
+
+			if (item != null && item.Parent == null)
+				return;
+
+			if (item != null)
+			{
+				var items = ((ContentPage)item.Parent).ToolbarItems;
```

Comment:
The parent could be any ```Page``` not just ```ContentPage``` so it would be better to cast to the least derived class that has a ```ToolbarItems``` property.

Reply:
Right, good point.